### DISCUSSION TITLE
FulfillmentInbound.GetShipments() problem with ShipmentStatus INTRANSIT or CHECKEDIN

### DISCUSSION THREAD

**Discussion by dave-sewell** [abuzuhri/Amazon-SP-API-CSharp#285] [Q&A] [answered]
This library has been a real time saver and relatively easy to use so far. However, I did run into a problem calling FulfillmentInbound.GetShipments() when using QueryType.SHIPMENT with ShipmentStatus.INTRANSIT or ShipmentStatus.CHECKEDIN. Here is the code illustrating the problem:

      ParameterGetShipments filter = new ParameterGetShipments();
      filter.MarketplaceId = marketplaceId;
      filter.ShipmentStatusList = new List<ShipmentStatus>();
      filter.ShipmentStatusList.Add(ShipmentStatus.WORKING);
      filter.ShipmentStatusList.Add(ShipmentStatus.SHIPPED);
      filter.ShipmentStatusList.Add(ShipmentStatus.RECEIVING);
      filter.ShipmentStatusList.Add(ShipmentStatus.DELIVERED);
	// The line below generates an "Invalid ShipmentStatus: INTRANSIT" error. Should be "IN_TRANSIT".
      filter.ShipmentStatusList.Add(ShipmentStatus.INTRANSIT);    
	// The line below generates an "Invalid ShipmentStatus: CHECKEDIN" error. Should be "CHECKED_IN".
      filter.ShipmentStatusList.Add(ShipmentStatus.CHECKEDIN);    
      filter.QueryType = Constants.QueryType.SHIPMENT;
      GetShipmentsResult shipmentsResult = amazonConnection.FulFillmentInbound.GetShipments(filter);

The ShipmentStatus enum needs to be fixed so that INTRANSIT is IN_TRANSIT and CHECKEDIN is CHECKED_IN.

Thanks.

**Comment by dave-sewell** [comment]
Update: the API documentation says to use "IN_TRANSIT" rather than "INTRANSIT", would would explain the error message noted. However, I noticed some actually get returned with "INTRANSIT" as the status, so not sure what is going on with that.

**Reply by abuzuhri** [reply]
This should be fixed in last version please check and let me know

**Comment by dave-sewell** [comment, marked as answer]
Yes, it works fine in version 1.4.23. Thanks for the quick fix!
